packages/runtime/src/prop-controllers/introspection.ts

```diff
@@ -303,7 +303,14 @@ export function getFileIds<T extends Data>(
 
     case Types.Images: {
       const value = prop as ImagesValue
-      return value?.flatMap(item => (item.props.file == null ? [] : [item.props.file])) ?? []
+      return value?.flatMap(
+        item => match(item.props.file)
+          .with(P.string, f => [f])
+          .with({type: 'makeswift-file', version: 1}, f => [f.id])
+          .with({type: 'external-file', version: 1}, () => [])
+          .with(P.nullish, () => [])
+          .exhaustive()
```

Comment:
Should we do `.otherwise(() => [])` here?

Reply:
It was more a question than a suggestion but since you did it I'm assuming the answer was "yes"?

Reply:
Ah my bad, the answer is that it shouldn't make a difference because it really should be exhaustive. I prefer using `.otherwise` because it allows for continuity of behavior in the strange case something does go wrong, however.